graviti/schema/pyarrow.py

```diff
@@ -93,3 +95,30 @@ def __init__(
 
     def _to_pyobj(self) -> Dict[str, Any]:
         return {**super()._to_pyobj(), "url": self._url, "revision": self._revision}
+
+
+class FileArray(  # pylint: disable=too-few-public-methods
+    pa.ExtensionArray  # type: ignore[misc]
+):
+    """This class defines the PyArrow representation of FileArray."""
+
+    def __getitem__(self, index: Union[int, slice]) -> Union[File, pa.Array]:
+        if isinstance(index, slice):
+            return super().__getitem__(index)
+
+        item = super().__getitem__(index).as_py()
+        return File(item["url"], item["checksum"])
+
+
+class FileType(BuiltinExtension):  # pylint: disable=too-few-public-methods
```

Comment:
`FileType` is `external` or `builtin`?

Reply:
The `ExternalExtension` has the neessary parameter `url`, which means refer to the other repo?